Src/PCompiler/CompilerCore/Backend/Stately/StatelyCodeGenerator.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Plang.Compiler.TypeChecker;
+using Plang.Compiler.TypeChecker.AST;
+using Plang.Compiler.TypeChecker.AST.Declarations;
+using Plang.Compiler.TypeChecker.AST.Statements;
+using Plang.Compiler.TypeChecker.AST.States;
+
+namespace Plang.Compiler.Backend.Stately {
+    public class StatelyCodeGenerator : ICodeGenerator
+    {
+        public bool HasCompilationStage => false;
+        public IEnumerable<CompiledFile> GenerateCode(ICompilerConfiguration job, Scope globalScope)
+        {
+            var context = new CompilationContext(job);
+            var statelySource = GenerateSource(context, globalScope);
+            return new List<CompiledFile> { statelySource };
+        }
+
+        private CompiledFile GenerateSource(CompilationContext context, Scope globalScope)
+        {
+            var source = new CompiledFile(context.FileName);
+            WriteSourcePrologue(context, source.Stream);
+            // write the top level declarations
+            foreach (var decl in globalScope.AllDecls)
+            {
+                WriteDecl(context, source.Stream, decl);
+            }
+
+            return source;
+        }
+
+        private void WriteDecl(CompilationContext context, StringWriter output, IPDecl decl)
+        {
+            string declName;
```

Comment:
Minor: Let's resolve this compiler warning: "The variable 'declName' is declared but never used"